validphys2/src/validphys/filters.py

```diff
@@ -179,9 +185,54 @@ def _filter_real_data(filter_path, data):
         dataset.load_commondata().export(path)
     return total_data_points, total_cut_data_points
 
+def process_commondata(commondata,ADD,MULT,CORR,UNCORR,inconsistent_datasets,sys_rescaling_factor):
+    """
+    Given a commondata instance return a commondata instance
+    with modified systematics. Note that if commondata.setname
+    is not within the inconsistent_datasets or if both ADD and
+    MULT are False, then the input commondata is outputted
+
+    Parameters
+    ----------
+    
+    commondata : validphys.coredata.CommonData
+
+    ADD : bool
+
+    MULT : bool
+
+    CORR : bool
+
+    UNCORR : bool
+
+    inconsistent_datasets : list
+                        list of the datasets for which an inconsistency should be introduced
+    
+    sys_rescaling_factor : float, int
+
+    Returns
+    -------
+    validphys.coredata.CommonData
+    """
+
+    if not commondata.setname in inconsistent_datasets:
+        return commondata
+    
+    if MULT:
+        commondata = commondata.with_MULT_sys(commondata.rescale_sys("MULT",CORR,UNCORR,sys_rescaling_factor))
+        
+
+    if ADD:
+        commondata = commondata.with_ADD_sys(commondata.rescale_sys("ADD",CORR,UNCORR,sys_rescaling_factor))
+
+    return commondata
 
 def _filter_closure_data(
-    filter_path, data, fakepdf, fakenoise, filterseed, experiments_index
+    filter_path, data, fakepdf, fakenoise, filterseed, experiments_index,
+    ADD=False, MULT=False, CORR=False, UNCORR=False,
+    inconsistent_datasets=[], sys_rescaling_factor_1=1, sys_rescaling_factor_2=1, 
```

Comment:
@giovannidecrescenzo, There is no need to have sys_rescaling_factor_1 and _2, it should be enough to have only one

Reply:
Yes I agree, but speaking with @andreab1997 it was also noticed that in this way one could perform a closure test which modifies both the covmats used (I do not see the need for it right now, but in this way there is at least the possibility)